katdal/h5datav2_5.py

```diff
@@ -0,0 +1,631 @@
+        sensors_group.visititems(register_sensor)
+        # Use estimated data timestamps for now, to speed up data segmentation
+        self.sensor = SensorCache(cache, data_timestamps, self.dump_period, keep=self._time_keep,
+                                  props=SENSOR_PROPS, virtual=VIRTUAL_SENSORS, aliases=SENSOR_ALIASES)
+
+        # ------ Extract subarrays ------
+        # By default, only pick antennas that were in use by the script
+        script_ants = str(config_group['Observation'].attrs['ants']).split(',')
+
+        self.ref_ant = script_ants[0] if not ref_ant else ref_ant
+        # Original list of correlation products as pairs of input labels
+        corrprods = get_single_value(config_group["DBE"], "vis_ordering").split(',')
+        if len(corrprods) != self._vis.shape[2]:
+            raise BrokenFile('Number of data labels (containing expected antenna names) '
+                             'received from h5 file (%d) differs from number of power products in data (%d)' %
+                             (len(corrprods), self._vis.shape[2]))
+        # Get the corrprod labels into the format that KatDAL wants, the v2.5 files only give ll and rr.
+        # This will change it into  "ant1lant1l,ant1rant1r" which katdal expects.
+        corrprods = [('ant1' + corrprods[0][0], 'ant1' + corrprods[0][1]),
+                     ('ant1' + corrprods[1][0], 'ant1' + corrprods[1][1])]
+
+        stokes_prods = get_single_value(config_group["DBE"], "stokes_ordering").split(',')
+        if len(stokes_prods) != self._stokes.shape[2]:
+            raise BrokenFile('Number of data labels (containing expected antenna names) '
+                             'received from h5 file (%d) differs from number of Stokes products in data (%d)' %
+                             (len(stokes_prods), self._stokes.shape[2]))
+        # All antennas in configuration as katpoint Antenna objects
+
+        ants = []
+        for antenna in config_group["Antennas"]:
+            name = config_group["Antennas"][antenna].attrs['name']
+            latitude = config_group["Antennas"][antenna].attrs['latitude']
+            longitude = config_group["Antennas"][antenna].attrs['longitude']
+            altitude = config_group["Antennas"][antenna].attrs['altitude']
+            diameter = config_group["Antennas"][antenna].attrs['diameter']
+            delay_model = None
+            pointing_model = make_pmodel_string(config_group["Antennas"][antenna]['pointing-model-params'])
+            beamwidth = config_group["Antennas"][antenna].attrs['beamwidth']
+
+            ants.append(katpoint.Antenna(name, latitude, longitude, altitude, diameter, delay_model, pointing_model, beamwidth))
+
+        # Temporarily casting this to string. I edited a file by hand to change the name and it seems to now read the
+        # attr as a np.ndarray
+        #ants = [katpoint.Antenna(str(config_group['Antennas'][name].attrs['description']))
```

Comment:
You should remove commented out code if not applicable anymore.

Reply:
Noted. Removed in 51587b5.